hack/generator/pkg/astmodel/identifier_factory.go

```diff
@@ -39,21 +47,31 @@ func NewIdentifierFactory() IdentifierFactory {
 }
 
 // CreateIdentifier returns a valid Go public identifier
-func (factory *identifierFactory) CreateIdentifier(name string) string {
+func (factory *identifierFactory) CreateIdentifier(name string, visibility Visibility) string {
 	if identifier, ok := factory.renames[name]; ok {
-		return identifier
+		name = identifier
 	}
 
 	// replace with spaces so titlecasing works nicely
 	clean := filterRegex.ReplaceAllLiteralString(name, " ")
 
-	titled := strings.Title(clean)
-	result := strings.ReplaceAll(titled, " ", "")
+	cleanWords := sliceIntoWords(clean)
+	var caseCorrectedWords []string
+	for i, word := range cleanWords {
+		if visibility == Internal && i == 0 {
+			caseCorrectedWords = append(caseCorrectedWords, strings.ToLower(word))
+		} else {
+			caseCorrectedWords = append(caseCorrectedWords, strings.Title(word))
+		}
+	}
+
+	result := strings.Join(caseCorrectedWords, "")
+	result = strings.ReplaceAll(result, " ", "")
```

Comment:
Do we need this? Under what circumstances would the `Join()` give us a string containing spaces?

Reply:
Likely never - can remove